packages/core/src/services/chains/run.ts

```diff
@@ -218,44 +231,33 @@ async function streamAIResult(
 
 async function handleCompletedChain(
   controller: ReadableStreamDefaultController,
-  stepResult: Awaited<ReturnType<typeof computeStepData>>,
-  response: ChainCallResponse,
+  stepResult: ValidatedStep,
+  response: ChainStepResponse<StreamType>,
 ) {
-  const eventData = {
-    type: ChainEventTypes.Complete,
-    config: stepResult.conversation.config as Config,
-    documentLogUuid: response.documentLogUuid,
-    response,
-  } as const
-
-  if ('text' in response) {
```

Comment:
This was the bug. In any case an AI response sent the same. It was always entering this `if` because an object response also has `.text` attribute. So the else condition was never used. 

What I did in my first attempt was to use my fancy `streamType` thinking this code worked as expeted
```javascript
if (response.streamType === 'text') {
} else if (response.streamType === 'object') {
  // Sending this instead of `messages` was the 🐛 
  Object.assign(eventData, {
    object: (response as ChainObjectResponse).object,
    text: objectToString((response as ChainObjectResponse).object),
  })
}
```

Reply:
Test test test test